packages/delegation-toolkit/src/actions/infuraBundlerClient.ts

```diff
@@ -0,0 +1,114 @@
+import type { Transport, Chain, Hex } from 'viem';
+import {
+  createBundlerClient,
+  type BundlerClientConfig,
+  type SmartAccount,
+} from 'viem/account-abstraction';
+
+/**
+ * Gas price tiers returned by pimlico_getUserOperationGasPrice
+ */
+export type GasPriceTier = {
+  /** Maximum fee per gas in hex format */
+  maxFeePerGas: Hex;
+  /** Maximum priority fee per gas in hex format */
+  maxPriorityFeePerGas: Hex;
+};
+
+/**
+ * Response from pimlico_getUserOperationGasPrice RPC method
+ */
+export type UserOperationGasPriceResponse = {
+  /** Slow gas price tier */
+  slow: GasPriceTier;
+  /** Standard gas price tier */
+  standard: GasPriceTier;
+  /** Fast gas price tier */
+  fast: GasPriceTier;
+};
+
+/**
+ * Infura bundler actions for extending bundler clients.
+ *
+ * @returns A function that takes a bundler client and returns the client extension with Infura bundler actions.
+ */
+const infuraBundlerActions = () => (client: any) => ({
+  /**
+   * Get user operation gas prices from Infura bundler.
+   * Calls the pimlico_getUserOperationGasPrice RPC method.
+   *
+   * @returns Promise resolving to gas price tiers (slow, standard, fast).
+   * @example
+   * ```typescript
+   * const gasPrices = await bundlerClient.getUserOperationGasPrice();
+   * console.log(gasPrices.standard.maxFeePerGas);
+   * ```
+   */
+  async getUserOperationGasPrice(): Promise<UserOperationGasPriceResponse> {
+    const response = await client.request({
```

Comment:
if you do a type assertion of the client, to include the pimlico bundler schema, the request method becomes type aware.

You've got to define the type schema, then fill in all of the optional generics on the client, so it gets a bit verbose, but I think is worthwhile for reducing risk of miss-calling.

```
type PimlicoBundlerSchema = [
  {
    Method: 'pimlico_getUserOperationGasPrices';
    Parameters: [];
    ReturnType: UserOperationGasPriceResponse;
  },
];

//...

const pimlicoClient = client as Client<
      Transport,
      Chain | undefined,
      Account | undefined,
      PimlicoBundlerSchema
>;

return await pimlicoClient.request({ method: 'pimlico_getUserOperationGasPrice', params: [] });
```